contracts/RestrictedNORI.sol

```diff
@@ -0,0 +1,1047 @@
+   *
+   * ##### Requirements:
+   *
+   * - the contract must not be paused
+   * - One of the following must be true:
+   *    - the operation is a mint (which should ONLY occur when BridgedPolygonNORI is being wrapped via `_depositFor`)
+   *    - the operation is a burn, which only happens during revocation and withdrawal:
+   *      - if the operation is a revocation, that permission is enforced by the TOKEN_REVOKER_ROLE
+   *      - if the operation is a withdrawal the burn amount must be <= the sender's claimable balance
+   *    - the operation is a transfer and _all_ of the following must be true:
+   *      - the operator is operating on their own balance (enforced in the inherited contract)
+   *      - the operator has sufficient balance to transfer (enforced in the inherited contract)
+   */
+  function _beforeTokenTransfer(
+    address operator,
+    address from,
+    address to,
+    uint256[] memory ids,
+    uint256[] memory amounts,
+    bytes memory data
+  ) internal override(ERC1155SupplyUpgradeable, ERC1155PausableUpgradeable) {
+    bool isBurning = to == address(0);
+    bool isWithdrawing = isBurning && from == operator;
+
+    if (isBurning) {
+      for (uint256 i = 0; i < ids.length; i++) {
+        uint256 id = ids[i];
+        Schedule storage schedule = _scheduleIdToScheduleStruct[id];
+        schedule.releasedAmountFloor = _releasedBalanceOfSingleSchedule(id);
+      }
+    }
+    if (isWithdrawing) {
+      for (uint256 i = 0; i < ids.length; i++) {
+        uint256 id = ids[i];
+        if (amounts[i] > claimableBalanceForScheduleForAccount(id, from)) {
+          revert InsufficientClaimableBalance({account: from, scheduleId: id});
+        }
+      }
+    }
+    return super._beforeTokenTransfer(operator, from, to, ids, amounts, data);
+  }
+
+  /**
+   * Linearly released balance for a single schedule at the current block timestamp, ignoring any
+   * released amount floor that has been set for the schedule.
+   */
+  /* solhint-disable not-rely-on-time, this is time-dependent */
+  function _linearReleaseAmountAvailable(uint256 scheduleId)
+    internal
+    view
+    returns (uint256)
+  {
+    Schedule storage schedule = _scheduleIdToScheduleStruct[scheduleId];
+    uint256 linearAmountAvailable;
+    // slither-disable-next-line timestamp this contract is block time dependent
+
+    if (block.timestamp >= schedule.endTime) {
+      linearAmountAvailable = totalSupply(scheduleId);
+    } else {
+      uint256 rampTotalTime = schedule.endTime - schedule.startTime;
+      // slither-disable-next-line timestamp this contract is block time dependent
+      linearAmountAvailable = block.timestamp < schedule.startTime
+        ? 0
+        : (_scheduleTrueTotal(schedule, scheduleId) *
+          (block.timestamp - schedule.startTime)) / rampTotalTime;
+    }
+    return linearAmountAvailable;
+  }
+
```

Comment:
```suggestion
```

Reply:
(locate the next line directly after the function in an attempt to try to keep the enable/disable located closer). Better yet, include it within the function definition itself instead

Reply:
the linter keeps auto-formatting this line into existence